src/services/resource.js

```diff
@@ -226,13 +234,115 @@ module.exports = class resourceHelper {
 		// generic function to merge all the collected data about the resource
 		result = await this.responseBuilder(response, userDetails, orgDetails, additionalResourceInformation)
 		// count of requested for changes resources
-		result.changes_requested_count = distinctInreviewResourceIds.count > 0 ? distinctInreviewResourceIds.count : 0
+		result.changes_requested_count = requestedForChangesResources.count > 0 ? requestedForChangesResources.count : 0
 		return responses.successResponse({
 			statusCode: httpStatusCode.ok,
 			message: 'RESOURCE_LISTED_SUCCESSFULLY',
 			result,
 		})
 	}
+
+	/**
+	 * Return the mapping of resource Id and its final status
+	 * @name finalResourceStatus
+	 * @param {Array} resourceId - Array of unique resource ids
+	 * @param {Array} organizationId - Array of unique org ids
+	 * @returns {JSON} - List of resources with final status
+	 */
+	static async finalResourceStatus(resourceId, organizationId) {
+		let recourceStatusMapping = {}
+
+		// Get the review details of all the resources created by the logged-in user
+		const resourceReviews = await reviewsQueries.findAll(
+			{
+				organization_id: {
+					[Op.in]: organizationId,
+				},
+				resource_id: {
+					[Op.in]: resourceId,
```

Comment:
@priyanka-TL this will never be multiple right, above also, using single eq operator